File: aula-9-class/aula-class.py
"""class Funcionario:
    def __init__(self, salario):
        self.salario = salario

    def getSalario(self):
        return self.salario

joao = Funcionario(2000)
print(joao.getSalario())"""



"""class Pessoa:
    def __init__(self, nome, idade):
        self.nome = nome
        self.idade = idade
    
    def getInfo(self):
        return self.nome, self.idade, self.cpf

class Pfisica(Pessoa):
    def __init__(self, nome, idade, cpf):
        super().__init__(nome, idade)
        self.cpf = cpf

joao = Pfisica("João", 17, "111.111.111-11")
print(joao.getInfo())"""






class Conta:
    def __init__(self, nome, numero):
        self.nome = nome
        self.numero = numero
        self.saldo = 0
    
    def depositar(self, valor):
        self.saldo += valor
    
    def saque(self, valor):
        if(self.saldo > 0 & self.saldo <= valor):
            self.saldo -= valor
        else:
            return "Dinhero insuficiente!"
    
    def getSaldo(self):
        return self.saldo

class ContaCorrente(Conta):
    def __init__(self, nome, numero):
        super().__init__(nome, numero)
    
    def depositar(self, valor):
        self.saldo += (valor - 2)
    
joao = ContaCorrente("João", 1111)
joao.depositar(200)
print(joao.getSaldo())
joao.saque(50)
print(joao.getSaldo())
joao.depositar(200)
print(joao.getSaldo())

File: aula-9-class/aula-class.js
/*class Pessoa {

    constructor(nome, cpf) {
        this.nome = nome;
        this.cpf = cpf
    }

    getNome() {
        return this.nome;
    }
}

var joao = new Pessoa("Teste", "111.111.111-11");
console.log(joao.getNome());*/

/*class Pessoa {
    constructor(nome, idade) {
        this.nome = nome;
        this.idade = idade;
    }
    
    showInfo(nome, idade) {
        return `${this.nome} tem ${this.idade}`
    }
}

class Pfisica extends Pessoa {
    constructor(nome, idade, cpf) {
        super(nome, idade);
        this.cpf = cpf;
    }
}

var pessoa1 = new Pfisica("João", 17, "111.111.111-11")
console.log(pessoa1);
console.log(pessoa1.showInfo())*/


class Conta{
    constructor(nome, numero) {
        this.nome = nome;
        this.numero = numero;
        this.saldo = 0;
    }

    deposito(valor) {
        this.valor = valor;
    }

    saque(valor) {
        if (this.valor > valor) {
            console.log("DINHEIRO INSUFICIENTE!");
        } else {
            this.saldo -= valor
        }
    }

    getSaldo() {
        return this.saldo;
    }
}

class Contacorrente extends Conta {
    constructor(nome, numero) {
        super(nome, numero);
    }

    deposito(valor) {
        this.saldo += (valor - 2);
    }
}

let joao = new Contacorrente("João", 123);
joao.deposito(200)
console.log(joao.getSaldo())
joao.saque(50)
console.log(joao.getSaldo())
joao.deposito(200)
console.log(joao.getSaldo())